src/test/java/codeu/controller/RegisterServletTest.java

```diff
@@ -30,4 +37,55 @@ public void testDoGet() throws IOException, ServletException {
     registerServlet.doGet(mockRequest,mockResponse);
     Mockito.verify(mockRequestDispatcher).forward(mockRequest,mockResponse);
   }
+  
+  @Test
+  public void testDoPost_BadUsername() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
+
+    registerServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest)
+        .setAttribute("error", "Please enter only letters, numbers, and spaces.");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoPost_NewUser() throws IOException, ServletException {
+    Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
+
+    UserStore mockUserStore = Mockito.mock(UserStore.class);
+    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(false);
+    registerServlet.setUserStore(mockUserStore);
+
+    HttpSession mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    registerServlet.doPost(mockRequest, mockResponse);
+
+    ArgumentCaptor<User> userArgumentCaptor = ArgumentCaptor.forClass(User.class);
+
+    Mockito.verify(mockUserStore).addUser(userArgumentCaptor.capture());
+    Assert.assertEquals(userArgumentCaptor.getValue().getName(), "test username");
+
+//    Mockito.verify(mockSession).setAttribute("user", "test username");
+//    Mockito.verify(mockResponse).sendRedirect("/conversations");
```

Comment:
The code they provided for you has a successful registration transition to the login page.  In that case you should verify the redirect is the conversations page, and the session attribute is not yet logged in.

Reply:
![image](https://user-images.githubusercontent.com/10622320/37633463-c069e904-2baf-11e8-9677-5f2545be85a8.png)
I set a password parameter but I am not user how to verify. I am confused about how set attribute works, does it mean I am assigning user and password values for mockSession? Also, do I need to add this one additional setattribute for password if I want to redirect to login page? the test is failing with the message below:
![image](https://user-images.githubusercontent.com/10622320/37633780-eeb6ec48-2bb0-11e8-8022-63b8d4bc9f63.png)
And I am also getting the same error for loginServletTest...